### ISSUE TITLE
Add support for SystemRDL source files

### ISSUE THREAD

**Issue by RasmusGOlsen** [edaa-org/pyEDAA.ProjectModel#47]
Can you add support for the register description language SystemRDL?

Here is a link to the standard https://www.accellera.org/downloads/standards/systemrdl.
There is also a great open-source community with a compiler and additional applications https://github.com/SystemRDL.

**Comment by Paebbels** [comment]
@RasmusGOlsen thanks for the interest in the project and proposing this feature. Also thanks for pull-request #48.

There I see a [pySRDLModel.py](https://github.com/edaa-org/pyEDAA.ProjectModel/pull/48/files#diff-8b8d5d6b56e7e87856c176170774ce8fbdcf116ff596d7aefe3c3e89ed336feb) file implementing at least the version numbers of SystemRDL.

This file would be shipped and installed underneath pyEDAA.ProjectModel, that's not where it should go...

Can can offer 2 solutions:
1. The GitHub organisation `SystemRDL` creates a `pySRDLModel` repository or
2. I can offer a `pySRDLModel` under the HDL organisation, or
3. I can also offer a `pySRDLModel` under the EDA² organisation.

I think the first solution is the better choice. The repository doesn't need to be complicated and could be a clone of [pySVModel](https://github.com/edaa-org/pySVModel). It would offer a separate repository and a matching PyPI package.

If you are an owner of `SystemRDL`, you could set me up as a maintainer for that repository and I'll take care of the initial setup from for code, documentation page, license, packaging etc.

**Comment by RasmusGOlsen** [comment]
Yes, I agree that the `pySRDLModel` should be its own repository. I'm not the owner of the SystemRDL organization, so  I don't have any mandate to create it. I can see that the `pyVHDLModel` is under the HDL organization and the `pySVModel` is under the EDA<sup>2</sup>. When I was searching for the models I first looked at the EDA<sup>2</sup> organization, for me that would make the most sense.

**Comment by Paebbels** [comment]
Lets create it at EDA² and propose a handover to SystemRDL. Can you create
an issue there?

RasmusGOlsen ***@***.***> schrieb am So., 18. Juni 2023,
21:39:

**Comment by Paebbels** [comment]
> I can see that the pyVHDLModel is under the HDL organization and the pySVModel is under the EDA2.

Just one tiny correction, `pyVHDLModel` is under `VHDL`, not `HDL` (but I have access to both organisations).

For `pySVModel` we don't have a host, so it's currently at EDA². Independent of the GitHub location, the package name at PyPI is reserved.

**Comment by Paebbels** [comment]
* [x] Repository created: https://github.com/edaa-org/pySystemRDLModel
* [x] Initial source code added (from pySVModel)
* [x] Initial documentation added: https://edaa-org.github.io/pySystemRDLModel/
* [x] Initial CI pipeline added: https://github.com/edaa-org/pySystemRDLModel/actions/runs/5306106193
* [ ] Modify code and unit test to suite SystemRDL.
* [ ] Review documentation to suite SystemRDL.
* [ ] Review README to suite SystemRDL.
* [ ] Exchange SV logo to SystemRDL logo.
* [ ] Add SystemRDL to EDA².
* [ ] First v0.1.0 release and registration at PyPI (needs my admin token)

**Comment by Paebbels** [comment]
#48 got merged.

**Comment by Paebbels** [comment]
I did some updates and bugfixes to SystemRDL introduced by copy&paste from SystemVerilog.
